File: LeetCode/101-200/121. 买卖股票的最佳时机.java
class Solution {
    public int maxProfit(int[] prices) {
  int maxP = 0,minPrice = Integer.MAX_VALUE;
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < minPrice) {
                minPrice = prices[i];
            } else if (prices[i] - minPrice > maxP) {
                maxP = prices[i] - minPrice;
            }
        }
        return maxP;
    }
}

File: LeetCode/101-200/121. 买卖股票的最佳时机.cpp
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int min_p = 1e9, max_p = 0;
        for (int p : prices)
        {
            max_p = max(max_p, p - min_p);
            min_p = min(min_p, p);
        }
        return max_p;
    }
};